src/Tribe/Repositories/Traits/Post_Attendees.php

```diff
@@ -0,0 +1,271 @@
+<?php
+/**
+ * Post attendees trait that contains all of the ORM filters that can be used for any repository.
+ *
+ * @since TBD
+ *
+ * @package Tribe\Tickets\Repositories\Traits
+ */
+
+namespace Tribe\Tickets\Repositories\Traits;
+
+/**
+ * Class Post_Attendees
+ *
+ * @since TBD
+ */
+trait Post_Attendees {
+
+	/**
+	 * Filters events to include only those that match the provided attendee(s).
+	 *
+	 * @since TBD
+	 *
+	 * @param int|array $attendee_ids The attendee(s) to filter by.
+	 */
+	public function filter_by_attendee( $attendee_ids ) {
+		$repo = $this;
+
+		// If the repo is decorated, use that.
+		if ( ! empty( $this->decorated ) ) {
+			$repo = $this->decorated;
+		}
+
+		global $wpdb;
+
+		$attendee_ids = (array) $attendee_ids;
+		$attendee_ids = array_map( 'absint', $attendee_ids );
+		$attendee_ids = array_unique( $attendee_ids );
+		$attendee_ids = implode( ', ', $attendee_ids );
+
+		$alias_attendee = 'ticket_attendee';
+
+		// Do all of the other filtering.
+		$this->filter_by_has_attendees( true );
+
+		// Add the by ID filter.
+		$repo->where_clause( "`{$alias_attendee}`.`ID` IN ( {$attendee_ids} )" );
+	}
+
+	/**
+	 * Filters events to include only those that match the provided attendee state.
+	 *
+	 * @since TBD
+	 *
+	 * @param bool $has_attendees Indicates if the event should have attendees attached to it or not.
+	 */
+	public function filter_by_has_attendees( $has_attendees = true ) {
+		$repo = $this;
+
+		// If the repo is decorated, use that.
+		if ( ! empty( $this->decorated ) ) {
+			$repo = $this->decorated;
+		}
+
+		global $wpdb;
+
+		$alias_event    = 'ticket_attendee_event';
+		$alias_attendee = 'ticket_attendee';
+
+		$event_meta_keys = method_exists( $this, 'attendee_to_event_keys' ) ? $this->attendee_to_event_keys() : [];
+		$event_meta_keys = array_map( [ $wpdb, '_real_escape' ], $event_meta_keys );
+		$event_meta_keys = '"' . implode( '", "', $event_meta_keys ) . '"';
+
+		$attendee_types = method_exists( $this, 'attendee_types' ) ? $this->attendee_types() : [];
+		$attendee_types = array_map( [ $wpdb, '_real_escape' ], $attendee_types );
+		$attendee_types = '"' . implode( '", "', $attendee_types ) . '"';
+
+		if ( $has_attendees ) {
+			// Join to the meta that relates attendees to events.
+			$repo->filter_query->join( "
+					LEFT JOIN `{$wpdb->postmeta}` AS `{$alias_event}`
+						ON `{$alias_event}`.`meta_value` = `{$wpdb->posts}`.`ID`
+				", $alias_event );
+
+			// Join to the meta that relates meta to attendees.
+			$repo->filter_query->join( "
+					LEFT JOIN `{$wpdb->posts}` AS `{$alias_attendee}`
```

Comment:
:no_entry_sign: Opening parenthesis of a multi-line function call must be the last content on the line

<sub>hash: 334cf3e25e22892ad1fca46ab7cbcbef</sub>